kedro/framework/cli/starters.py

```diff
@@ -355,16 +355,28 @@ def _create_project(template_path: str, cookiecutter_args: Dict[str, str]):
         ) from exc
 
     _clean_pycache(Path(result_path))
+    extra_context = cookiecutter_args.get("extra_context", {})
+    project_name = extra_context.get("project_name", "")
+    python_package = extra_context.get(
+        "python_package", project_name.lower().replace(" ", "_").replace("-", "_")
+    )
```

Comment:
Fair point! I guess I was just being overly cautious but it might be better if this throws an error if there's no `extra_context` because then something would be really wrong.

Reply:
Oohh and actually I forgot about this, but we added in backward compatibility for when there's no `prompts.yml` file. Then you can still run `kedro new` and it will create a new project with "New Kedro Project" as project name which is the default in `cookiecutter.json`. So `project_name = extra_context["project_name"]` would break that and we should instead do extra_context.get("project_name", "New Kedro Project")

Reply:
Thanks for checking this! I think I understand. We should definitely see if we can get rid of the prompts.yml file though, it is pretty confusing 😬